flakytests/execution-time-limit/00-background.t

```diff
@@ -34,7 +34,7 @@ suite_run_fail "${TEST_NAME_BASE}-run" \
 
 LOGD="$(cylc get-global-config --print-run-dir)/${SUITE_NAME}/log/job/1/foo"
 grep_ok '# Execution time limit: 5.0' "${LOGD}/01/job"
-grep_ok 'CYLC_JOB_EXIT=\(ERR\|XCPU\)' "${LOGD}/01/job.status"
+grep_ok 'CYLC_JOB_EXIT=XCPU' "${LOGD}/01/job.status"
```

Comment:
This test is a manifestation of unreliability of trap execution order in bash. After this PR we should consistently see an exit via signal trap.